tests/parametric/test_otel_span_methods.py

```diff
@@ -71,6 +71,7 @@ def test_otel_set_service_name(self, test_agent, test_library):
         context.library == "java",
         reason="Old array encoding was removed in 1.22.0 and new span naming introduced in 1.24.0: no version elligible for this test.",
     )
+    @irrelevant(context.library == "golang", reason="Old array encoding no longer supported")
```

Comment:
We could add since version that this feature is irrelevant

@irrelevant(context.library < "golang@1.58.0", reason="Old array encoding no longer supported")

Reply:
@katiehockman ^ its worth adding the library version information, as we're not going to have false results - when testing older versions.

Reply:
Sounds good. Done.

Reply:
This actually needed to be `< "golang@1.59.0"` since the changes haven't been released yet.
Correcting this in a follow-up PR.

Reply:
https://github.com/DataDog/system-tests/pull/1885